Source/cairo/WaylandCairoEglSurface.m

```diff
@@ -0,0 +1,350 @@
+                window->width, window->height);
+
+        if (surface == NULL) {
+            NSDebugLog(@"can't create cairo surface");
+        }
+        return surface;
+}
+
+
+@implementation WaylandCairoEglSurface
+{
+    struct ESContext _escontext;
+    unsigned char * memoryBuff;
+    cairo_device_t * _cairo_device;
+    cairo_surface_t * _eglSurface;
+    cairo_surface_t * _memSurface;
+}
+- (id) initWithDevice: (void*)device
+{
+
+    gsDevice = device;
+
+    struct window *window = (struct window *) device;
+
+    _escontext.native_display = window->wlconfig->display;
+    _escontext.window_width = window->width;
+    _escontext.window_height = window->height;
+    _escontext.display = NULL;
+    _escontext.context = NULL;
+    _escontext.surface = NULL;
+    _escontext.native_window  = 0;
+
+    InitEGLContext(&_escontext);
+
+    NSDebugLog(@"WaylandCairoSurface: initWithDevice win=%d", window->window_id);
+
+    _cairo_device = cairo_egl_device_create(_escontext.display, _escontext.context);
+
+    // If your application is not multithreading, add cairo_gl_device_set_thread_aware (device, FALSE) to your code to reduce context switches
+    //
+    // FIXME setting this to false breaks everything
+    // cairo_gl_device_set_thread_aware(_cairo_device, FALSE);
+
+    _memSurface = cairo_gl_surface_create(_cairo_device, CAIRO_CONTENT_COLOR_ALPHA, window->width, window->height);
+
+    // the egl surface is going to be created when we need to display the surface on screen
+    _eglSurface = NULL;
+
+    _surface = _memSurface;
+
+    window->wcs = self;
+
+    return self;
+}
+
+- (id) initLayer:(void*)owner withDevice: (void*)device
+{
+    struct window *window = (struct window *) device;
+    return [self initWithDevice: device];
+}
+
+- (void) dealloc
+{
+    struct window *window = (struct window*) gsDevice;
+    NSDebugLog(@"WaylandCairoEglSurface: dealloc win=%d", window->window_id);
+
+    // destroy egl
+    if(_eglSurface) {
+
+        eglDestroySurface(_escontext.display, _escontext.surface);
+        wl_egl_window_destroy(_escontext.native_window);
+    }
+
+    [super dealloc];
+}
+
+- (NSSize) size
+{
+    struct window *window = (struct window*) gsDevice;
+    return NSMakeSize(window->width, window->height);
+}
+
+- (void) setSize: (NSSize)newSize
+{
+    struct window *window = (struct window*) gsDevice;
+    NSDebugLog(@"[%d] cairo surface set size", window->window_id);
+
+}
+- (BOOL) isDrawingToScreen
+{
+    return _eglSurface != NULL;
+}
+
+- (void) handleExposeRect: (NSRect)rect
+{
+    struct window *window = (struct window*) gsDevice;
+
+    int x = NSMinX(rect);
+    int y = NSMinY(rect);
+    int width = NSWidth(rect);
+    int height = NSHeight(rect);
+    window->buffer_needs_attach = YES;
+    if (window->configured) {
+        // if the window is configured, it's time to display the content on screen
+        // we are creating a window surface
+        if(_eglSurface == NULL) {
```

Comment:
Again this does not follow the GNUstep coding guidelines.

Reply:
thanks for reviewing it, i planned to refactor the code after rebasing with the other PR

Reply:
in here the refactored code following the guidelines
https://github.com/nongio/libs-back/pull/2